### PR TITLE
planner: add explain for InspectionResultTableExtractor

### LINKED ISSUE #15381: UCP: add explain for InspectionResultTableExtractor
## Description

This is a sub-task of https://github.com/pingcap/tidb/issues/15378.

What need todo? Implement the `explainInfo` metheod of below extractor.

- [ ] InspectionResultTableExtractor

## Score

* 50 score.

## Mentor(s)

* [@crazycs](https://github.com/crazycs520)

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

## Recommended Skills

* go

## Learning Materials

* Chinese: [TiDB 精选技术讲解文章](https://github.com/pingcap/presentations/blob/master/hackathon-2019/reference-document-of-hackathon-2019.md)
* English: [Awesome-Database-Learning](https://github.com/pingcap/awesome-database-learning)

### PR DESCRIPTION
**Pull request by LindaSummer**
### What problem does this PR solve?

Issue Number: close #15381 

What's Changed:
Add explain info for InspectionResultTableExtractor
1.  planner/core: memtable_predicate_extractor.go
2. *: Add explain info for InspectionResultTableExtractor

Here is an example I query it locally:
```
mysql> desc select * from information_schema.inspection_result where item in ('ddl.lease', 'raftstore.threadpool') and rule in ('ddl', 'config');
+----------------+----------+------+-------------------------+-------------------------------------------------------------------+
| id             | estRows  | task | access object           | operator info                                                     |
+----------------+----------+------+-------------------------+-------------------------------------------------------------------+
| MemTableScan_5 | 10000.00 | root | table:INSPECTION_RESULT | rules:["config","ddl"],items:["ddl.lease","raftstore.threadpool"] |
+----------------+----------+------+-------------------------+-------------------------------------------------------------------+
1 row in set (0.07 sec)
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- none

### CODE REVIEW

**Comment by LindaSummer** [comment]
run-codecov

**Comment by LindaSummer** [comment]
run-all-tests

**Comment by LindaSummer** [comment]
@qw4990 @winoros PTAL

File: planner/core/memtable_predicate_extractor.go [hunk 1]
```diff
@@ -828,7 +828,13 @@ func (e *InspectionResultTableExtractor) Extract(
 }
 
 func (e *InspectionResultTableExtractor) explainInfo(p *PhysicalMemTable) string {
-	return ""
+	if e.SkipInspection {
+		return "skip_inspection: true"
```
Comment (zz-jason) [inline planner/core/memtable_predicate_extractor.go:832]:
```suggestion
		return "skip_inspection:true"
```

to be compatible with others

File: planner/core/memtable_predicate_extractor.go [hunk 2]
```diff
@@ -828,7 +828,13 @@ func (e *InspectionResultTableExtractor) Extract(
...
+	}
+	s := make([]string, 0, 2)
+	s = append(s, fmt.Sprintf("rules:[%s]", extractStringFromStringSet(e.Rules)))
+	s = append(s, fmt.Sprintf("items:[%s]", extractStringFromStringSet(e.Items)))
+	return strings.Join(s, ",")
```
Comment (qw4990) [inline planner/core/memtable_predicate_extractor.go:837]:
```suggestion
	return strings.Join(s, ", ")
```

**Comment by qw4990** [comment]
@LindaSummer Sorry for our late reply. It's almost LGTM, please address comments and resolve conflicts.

File: executor/explainfor_test.go [hunk 3]
```diff
@@ -140,6 +140,18 @@ func (s *testSuite) TestExplainClusterTable(c *C) {
 		`MemTableScan_5 10000.00 root table:CLUSTER_CONFIG node_types:["tidb"], instances:["192.168.1.7:2379"]`))
 }
 
+func (s *testSuite) TestInspectionResultTable(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustQuery("desc select * from information_schema.inspection_result where rule = 'ddl' and rule = 'config'").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT skip_inspection: true`))
```
Comment (crazycs520) [inline executor/explainfor_test.go:146]:
```suggestion
		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT skip_inspection:true`))
```

File: executor/explainfor_test.go [hunk 4]
```diff
@@ -140,6 +140,18 @@ func (s *testSuite) TestExplainClusterTable(c *C) {
 		`MemTableScan_5 10000.00 root table:CLUSTER_CONFIG node_types:["tidb"], instances:["192.168.1.7:2379"]`))
 }
 
+func (s *testSuite) TestInspectionResultTable(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustQuery("desc select * from information_schema.inspection_result where rule = 'ddl' and rule = 'config'").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT skip_inspection:true`))
+	tk.MustQuery("desc select * from information_schema.inspection_result where rule in ('ddl', 'config')").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:["config","ddl"],items:[]`))
```
Comment (crazycs520) [inline executor/explainfor_test.go:148]:
```suggestion
		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:["config","ddl"], items:[]`))
```

File: executor/explainfor_test.go [hunk 5]
```diff
@@ -140,6 +140,18 @@ func (s *testSuite) TestExplainClusterTable(c *C) {
...
 }
 
+func (s *testSuite) TestInspectionResultTable(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustQuery("desc select * from information_schema.inspection_result where rule = 'ddl' and rule = 'config'").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT skip_inspection:true`))
+	tk.MustQuery("desc select * from information_schema.inspection_result where rule in ('ddl', 'config')").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:["config","ddl"], items:[]`))
+	tk.MustQuery("desc select * from information_schema.inspection_result where item in ('ddl.lease', 'raftstore.threadpool')").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:[],items:["ddl.lease","raftstore.threadpool"]`))
```
Comment (crazycs520) [inline executor/explainfor_test.go:150]:
```suggestion
		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:[], items:["ddl.lease","raftstore.threadpool"]`))
```

File: executor/explainfor_test.go [hunk 6]
```diff
@@ -140,6 +140,18 @@ func (s *testSuite) TestExplainClusterTable(c *C) {
...
+func (s *testSuite) TestInspectionResultTable(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustQuery("desc select * from information_schema.inspection_result where rule = 'ddl' and rule = 'config'").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT skip_inspection:true`))
+	tk.MustQuery("desc select * from information_schema.inspection_result where rule in ('ddl', 'config')").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:["config","ddl"], items:[]`))
+	tk.MustQuery("desc select * from information_schema.inspection_result where item in ('ddl.lease', 'raftstore.threadpool')").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:[], items:["ddl.lease","raftstore.threadpool"]`))
+	tk.MustQuery("desc select * from information_schema.inspection_result where item in ('ddl.lease', 'raftstore.threadpool') and rule in ('ddl', 'config')").Check(testkit.Rows(
+		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:["config","ddl"],items:["ddl.lease","raftstore.threadpool"]`))
```
Comment (crazycs520) [inline executor/explainfor_test.go:152]:
```suggestion
		`MemTableScan_5 10000.00 root table:INSPECTION_RESULT rules:["config","ddl"], items:["ddl.lease","raftstore.threadpool"]`))
```

**Pull request merged**